charts/devnet/templates/chains/virtual/eth/statefulset.yaml

```diff
@@ -0,0 +1,168 @@
+          image: {{ $chain.geth.image }}
+          command: ['sh', '-c', "if [ -e /ethereum/.init ]; then exit 0; fi; geth init --datadir=/ethereum/execution /etc/config/genesis.json"]
+          volumeMounts:
+            - name: ethereum
+              mountPath: /ethereum
+            - name: config
+              mountPath: /etc/config
+              readOnly: true
+        - name: init-done
+          image: alpine/openssl
+          command: ['sh', '-c', "touch /ethereum/.init"]
+          volumeMounts:
+            - name: ethereum
+              mountPath: /ethereum
+      containers:
+        - name: geth
+          image: {{ $chain.geth.image }}
+          args:
+            {{- range $arg := $chain.geth.args }}
+            - {{ $arg -}}
+            {{ end }}
+          workingDir: /ethereum/execution
+          ports:
+            - name: geth-http
+              containerPort: 8545
+            - name: geth-authrpc
+              containerPort: 8551
+          volumeMounts:
+            - name: ethereum
+              mountPath: /ethereum
+            - name: secrets
+              mountPath: /etc/secrets
+              readOnly: true
+          resources:
+            requests:
+              cpu: 100m
+              memory: 1Gi
+        - name: beacon-chain
+          image: {{ $chain.beaconChain.image }}
+          command: {{ $chain.beaconChain.command }}
+          args:
+            {{- range $arg := $chain.beaconChain.args }}
+            - {{ $arg -}}
+            {{ end }}
+          workingDir: /ethereum/consensus
+          ports:
+            - name: beacon-api
+              containerPort: 3500
+            - name: beacon-grpc
+              containerPort: 4000
+            - name: beacon-web
+              containerPort: 8080
+          volumeMounts:
+            - name: ethereum
+              mountPath: /ethereum
+            - name: config
+              mountPath: /etc/config
+              readOnly: true
+            - name: secrets
+              mountPath: /etc/secrets
+              readOnly: true
+          resources:
+            requests:
+              cpu: 100m
+              memory: 1Gi
+        - name: validator
+          image: {{ $chain.validator.image }}
+          command: {{ $chain.validator.command }}
+          args:
+            {{- range $arg := $chain.validator.args }}
+            - {{ $arg -}}
+            {{ end }}
+          workingDir: /ethereum/consensus
+          volumeMounts:
+            - name: ethereum
+              mountPath: /ethereum
+            - name: config
+              mountPath: /etc/config
+              readOnly: true
+          resources:
+            requests:
+              cpu: 100m
+              memory: 1Gi
+      {{- include "imagePullSecrets" $chain | indent 6 }}
+      volumes:
+        - name: secrets
+          secret:
+            secretName: geth
+        - name: config
+          configMap:
+            name: eth2
+  volumeClaimTemplates:
```

Comment:
Any reasons to use this instead of just empty dir? Are you deploying this somewhere where you were having issues with empty dir?

We would have to change the resource request, or make this part of the config where one can specify the volume, else this template wont be able to run well locally.

Reply:
I am using a volume claim template to allocate disk space so the eth nodes in case of a restart or any issues won't lose any blockchain data they've accumulated/created so far. I am not a k8s expert, let me know if `emptyDir` can be used in this case.

Also we plan to use starship to run polymer alpha testnet in GKE and I am not sure how `emptyDir` behaves in a cloud environment.

> We would have to change the resource request, or make this part of the config where one can specify the volume, else this template wont be able to run well locally.
Can you explain what is the problem with running well locally?

Reply:
In cloud environment this makes sense, but not in other environments (if someone wants to spinup eth chains locally). Our goal should be, ability to run this helm chart basically in any environment. Maybe we need a redesign of sorts, so that this helm chart is better used in a cloud setting.

Oh so you are using this helm chart directly for your testnets?